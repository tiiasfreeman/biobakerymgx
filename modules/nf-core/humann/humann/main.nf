process HUMANN_HUMANN {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/humann:3.8--pyh7cba7a3_0':
        'biocontainers/humann:3.8--pyh7cba7a3_0' }"

    input:
    tuple val(meta), path(fastq)
    path             (chocophlan_db_dir)
    path             (uniref_db_dir)

    output:
    tuple val(meta), path("*._genefamilies.tsv"),  emit: genefamilies
    tuple val(meta), path("*._pathcoverage.tsv"),  emit: pathcoverage
    tuple val(meta), path("*._pathabundance.tsv"), emit: pathabundance
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    humann \\
    --input ${fastq} \\
    --nucleotide-database $chocophlan_db_dir \\
    --protein-database $uniref_db_dir \\
    --output humann_results \\
    ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        modules: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    touch "*._genefamilies.tsv"
    touch "*._pathcoverage.tsv"
    touch "*._pathabundance.tsv"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        modules: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
}