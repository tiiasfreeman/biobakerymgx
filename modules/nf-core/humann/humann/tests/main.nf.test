nextflow_process {

    name "Test Process HUMANN_HUMANN"
    script "/workspace/biobakerymgx/modules/nf-core/humann/humann/main.nf"
    process "HUMANN_HUMANN"

    tag "modules"
    tag "modules_nfcore"
    tag "humann"
    tag "humann/humann"

    setup {
        run("HUMANN_DOWNLOADCHOCOPHLANDB") {
            script "/workspace/biobakerymgx/modules/nf-core/humann/downloadchocophlandb/main.nf"
        }

        run("HUMANN_DOWNLOADUNIREFDB") {
            script "/workspace/biobakerymgx/modules/nf-core/humann/downloadunirefdb/main.nf"
        }
    }

    test("bacteroides_fragilis - illumina - test1_contigs_fa_gz - stub") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test_fastq' ],
                        file("https://github.com/biobakery/humann/raw/master/examples/demo.fastq.gz", checkIfExists: true)
                    ])
                input[1] = HUMANN_DOWNLOADCHOCOPHLANDB.out.chocophlan_db_dir
                input[2] = HUMANN_DOWNLOADUNIREFDB.out.uniref_db_dir
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
