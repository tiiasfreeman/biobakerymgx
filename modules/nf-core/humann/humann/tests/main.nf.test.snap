{
    "HUMANN_HUMANN": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "genefamilies": [
                    
                ],
                "pathabundance": [
                    
                ],
                "pathcoverage": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T21:15:13.599763794"
    },
    "bacteroides_fragilis - illumina - test1_contigs_fa_gz - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "genefamilies": [
                    
                ],
                "pathabundance": [
                    
                ],
                "pathcoverage": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T21:10:56.13956366"
    }
}